File: eecs168/2021summer/2021.07.20.arrays-of-objects-part-2/Triangle.cpp
//Triangle.cpp

#include "Triangle.h"
/*
Triangle::Triangle()
{
	//Initialize your member variables
	base = 0;
	height = 0;
}
*/
Triangle::Triangle(double newBase, double newHeight)
{
	//Where's the object? 
	//We're in the object!
	setBase(newBase);
	setHeight(newHeight);
}

double Triangle::getBase() const 
{
	return(base);
}

double Triangle::getHeight() const
{
	return(height);	
}

bool Triangle::setBase(double newBase)
{
	if(newBase > 0)
	{
		base = newBase;
		return(true);
	}
	else
	{
		base = 0;
		return(false);
	}
}

bool Triangle::setHeight(double newHeight)
{
	if(newHeight > 0)
	{
		height = newHeight;
		return(true);
	}
	else
	{
		height = 0;
		return (false);
	}
}

double Triangle::area() const
{
	return(0.5*base*height);
}

File: eecs168/2025spring/MWF/2025.04.21.classes-part-4/triangle/triangle.py
#triangle.py
class Triangle:
    def __init__(self, base, height):
        print("Triangle's init called")
        #member variables
        self.base = base
        self.height = height

    def area(self):
        ans = 0.5*self.base*self.height
        return ans

    def __gt__(self, other):
        print('__gt__ called')
        if self.area() > other.area():
            return True
        else:
            return False
